internal/cluster/distribution/eks/eksworkflow/activity_update_node_group.go

```diff
@@ -194,6 +197,14 @@ func (a UpdateNodeGroupActivity) Execute(ctx context.Context, input UpdateNodeGr
 		},
 	}
 
+	if input.TemplateVersion != "1.0.0" {
+		stackParams = append(stackParams, sdkCloudFormation.NewOptionalStackParameter(
+			"CustomNodeSecurityGroups",
+			input.SecurityGroups != nil,
+			strings.Join(input.SecurityGroups, ","),
+		))
+	}
```

Comment:
**Required**: we also **MUST** provide the value if explicitly set, even if the stack is older than the introduction, so there should be an else branch for specified argument with explicit stack parameter setting.

```golang
} else if input.SecurityGroups != nil {
    stackParams = append(stackParamse, &cloudformation.Parmeter{
        ParameterKey: "CustomNodeSecurityGroups",
        ParameterValue: strings.Join(input.SecurityGroups, ","),
    })
}
```

Also as an idea, I would think about whether `NewOptionalStackParameter` can be extended with a version restriction to only use `UsePreviousValue` if a version requirement is met, because this feels to be a common case moving forward with versioned templates.

Reply:
Fixed